pkg/kritis/cryptolib/verifier.go

```diff
@@ -70,23 +81,32 @@ func NewVerifier(imageDigest string, publicKeySet []PublicKey) (Verifier, error)
 // VerifyAttestation verifies an Attestation. See Verifier for more details.
 func (v *verifier) VerifyAttestation(att *Attestation) error {
 	var (
-		err     error
-		payload []byte
+		err       error
+		payload   []byte
+		publicKey PublicKey
 	)
 
 	// Extract the public key from `publicKeySet` whose ID matches the one in
 	// `att`.
-	// TODO: Replace no-op with correct implementation.
-	publicKey := v.PublicKeySet[0]
+	foundKey := false
+	for _, key := range v.PublicKeySet {
+		if key.ID == att.PublicKeyID {
+			publicKey, foundKey = key, true
+			break
+		}
+	}
+	if !foundKey {
+		return fmt.Errorf("Verifier doesn't contain matching public key with ID %s", att.PublicKeyID)
```

Comment:
This will probably be a user visible error message, so make sure its text doesn't contain anything which requires understanding Kritis internals ("Verifier" for example is an internal term).  Maybe just: "no public key with ID %s found" or something,

Reply:
Done.